core/src/main/kotlin/io/github/paulgriffith/kindling/core/ToolPanel.kt

```diff
@@ -23,31 +23,36 @@ abstract class ToolPanel(
 
     open fun customizePopupMenu(menu: JPopupMenu) = Unit
 
-    protected fun exportMenu(modelSupplier: () -> TableModel): JMenu = JMenu("Export").apply {
-        for (format in ExportFormat.values()) {
-            add(
-                Action("Export as ${format.extension.uppercase()}") {
-                    exportFileChooser.resetChoosableFileFilters()
-                    exportFileChooser.fileFilter = format.fileFilter
-                    if (exportFileChooser.showSaveDialog(this.parent.parent) == JFileChooser.APPROVE_OPTION) {
-                        val selectedFile = if (exportFileChooser.selectedFile.endsWith(format.extension)) {
-                            exportFileChooser.selectedFile
-                        } else {
-                            File(exportFileChooser.selectedFile.absolutePath + ".${format.extension}")
+    protected fun exportMenu(defaultFileName: String = "", modelSupplier: () -> TableModel): JMenu =
+        JMenu("Export").apply {
+            for (format in ExportFormat.values()) {
+                add(
+                    Action("Export as ${format.extension.uppercase()}") {
+                        exportFileChooser.selectedFile = File(defaultFileName)
+                        exportFileChooser.resetChoosableFileFilters()
+                        exportFileChooser.fileFilter = format.fileFilter
+                        if (exportFileChooser.showSaveDialog(this.parent.parent) == JFileChooser.APPROVE_OPTION) {
+                            val selectedFile =
+                                if (exportFileChooser.selectedFile.absolutePath.endsWith(format.extension)) {
+                                    exportFileChooser.selectedFile
+                                } else {
+                                    File(exportFileChooser.selectedFile.absolutePath + ".${format.extension}")
+                                }
+                            format.action.invoke(modelSupplier(), selectedFile)
```

Comment:
Could clean this up some/reduce duplication by using `exportFileChooser.apply {`. I know the original code doesn't use that, but looking over it again I think it'd be an improvement.